demoapp/extract_mgl_sections.py

```diff
@@ -0,0 +1,319 @@
+import pandas as pd
+import re
+import numpy as np
+import requests
+import urllib3
+import requests
+from requests.exceptions import RequestException
+from requests.packages.urllib3.exceptions import InsecureRequestWarning
+import logging
+
+logging.basicConfig(level=logging.WARNING, format='%(asctime)s - %(levelname)s - %(message)s')
+
+def csv_to_df_cleaned(bills_csv_file_path, bill_title_col_name, bill_num_col_name, docket_num_col_name, bill_text_col_name):
+    """
+      Parameters:
+    - bills_csv_file_path (str): Path to the CSV file containing columns "Title", "BillNumber", "DocketNumber", "DocumentText".
+    - bill_title_col_name (str): Name of the column containing bill titles.
+    - bill_num_col_name (str): Name of the column containing bill numbers.
+    - docket_num_col_name (str): Name of the column containing docket numbers.
+    - bill_text_col_name (str): Name of the column containing bill text.
+
+    Returns:
+    - cleaned_df (pd.DataFrame): Modified Pandas DataFrame containing selected columns and an additional "Bill Info" column
+      with the combined Bill Number and Title for demo_app.py.
+    """
+    try: 
+        df = pd.read_csv(bills_csv_file_path)
+        # logging.info('CSV file read successfully')
+        
+        cleaned_df = df[[bill_title_col_name, bill_num_col_name, docket_num_col_name, bill_text_col_name]]
+        
+        #Adding a column with the Bill Number and Title combined to be used for the demo_app.py
+        bill_info = df[bill_num_col_name] + ": " + df[bill_title_col_name]
+        cleaned_df.insert(0, 'Bill Info', bill_info)
+        
+        return cleaned_df
+    
+    except FileNotFoundError:
+        logging.error('Error: The file %s was not found', bills_csv_file_path)
+        return None 
+
+def extract_sections(bill_text):
```

Comment:
Is this the correct output type?

```suggestion
def extract_sections(bill_text: str) -> list[tuple[str, str]]:
```

Reply:
I see you implemented a less detailed type, i.e., `def extract_sections(bill_text: str) -> list: `

Is the full type I suggested above more accurate?

Reply:
Yeah, you are right. It is a list of lists. So, your suggestion is more accurate.